apps/web/src/pages/api/v1/[...ts-rest].tsx

```diff
@@ -0,0 +1,233 @@
+
+    try {
+      const document = await getDocumentById({ id: Number(documentId), userId: user.id });
+
+      return {
+        status: 200,
+        body: document,
+      };
+    } catch (e) {
+      return {
+        status: 404,
+        body: {
+          message: 'Document not found',
+        },
+      };
+    }
+  },
+  deleteDocument: async (args) => {
+    const { id: documentId } = args.params;
+    const { authorization } = args.headers;
+
+    let user;
+
+    try {
+      user = await checkUserFromToken({ token: authorization });
+    } catch (e) {
+      return {
+        status: 401,
+        body: {
+          message: e.message,
+        },
+      };
+    }
+
+    try {
+      const document = await getDocumentById({ id: Number(documentId), userId: user.id });
+
+      const deletedDocument = await deleteDocument({
+        id: Number(documentId),
+        userId: user.id,
+        status: document.status,
+      });
+
+      return {
+        status: 200,
+        body: deletedDocument,
+      };
+    } catch (e) {
+      return {
+        status: 404,
+        body: {
+          message: 'Document not found',
+        },
+      };
+    }
+  },
+  createDocument: async (args) => {
+    const { body } = args;
+
+    try {
+      const { url, key } = await getPresignPostUrl(body.fileName, body.contentType);
+
+      return {
+        status: 200,
+        body: {
+          url,
+          key,
+        },
+      };
+    } catch (e) {
+      return {
+        status: 404,
+        body: {
+          message: 'An error has occured while uploading the file',
+        },
+      };
+    }
+  },
+  sendDocumentForSigning: async (args) => {
+    const { authorization } = args.headers;
+    const { id } = args.params;
+    const { body } = args;
+    let user;
+
+    try {
+      user = await checkUserFromToken({ token: authorization });
+    } catch (e) {
+      return {
+        status: 401,
+        body: {
+          message: e.message,
+        },
+      };
+    }
+
+    const document = await getDocumentById({ id: Number(id), userId: user.id });
+
+    if (!document) {
+      return {
+        status: 404,
+        body: {
+          message: 'Document not found',
+        },
+      };
+    }
+
+    if (document.status === 'PENDING') {
+      return {
+        status: 400,
+        body: {
+          message: 'Document is already waiting for signing',
+        },
+      };
+    }
+
+    try {
+      await setRecipientsForDocument({
+        userId: user.id,
+        documentId: Number(id),
+        recipients: [
+          {
+            email: body.signerEmail,
+            name: body.signerName ?? '',
+          },
+        ],
+      });
+
+      await setFieldsForDocument({
+        documentId: Number(id),
+        userId: user.id,
+        fields: body.fields.map((field) => ({
+          signerEmail: body.signerEmail,
+          type: field.fieldType,
+          pageNumber: field.pageNumber,
+          pageX: field.pageX,
```

Comment:
I'm not so sure about these `page` values. Is there a better way to specify where to place the field and how small/large to make it?

Reply:
The values are percentage based and are the simplest way of specifying the coordinates, v0.9 used absolute x,y values which didn't work well unless you intimately knew the document.

Ideally we will improve upon this later by reading up forms and annotations from the PDF to use as targets for fields

Reply:
Got it! Was just curious if there is a way to make it more intuitive/easier for the user.